.github/workflows/containers.yaml

```
@@ -149,9 +144,59 @@ jobs:
         with:
           file: containers/Dockerfile.${{ matrix.container.name }}
           pull: true  # ensure we always have an up to date source
-          push: ${{ github.event_name != 'pull_request' }}
+          push: true
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
           annotations: ${{ steps.meta.outputs.annotations }}
           # prod is the target that has the code installed
           target: prod
+
+  clean_containers:
+    if: github.event_name == 'pull_request' && github.event.action == 'closed'
+    runs-on: ubuntu-latest
+
+    permissions:
+      packages: write
+
+    strategy:
+      matrix:
+        container:
+          - ironic
+          - neutron
+          - dnsmasq
+          - ironic-nautobot-client
+
+    steps:
+      - name: clean up PR container
+        uses: actions/github-script@v7
+        env:
+          CONTAINER_NAME: '${{ matrix.container }}'
+        with:
+          script: |
+            const container_name = `understack%2F${process.env.CONTAINER_NAME}`;
+            const response = github.rest.packages.getAllPackageVersionsForPackageOwnedByOrg({
```

The `getAllPackageVersionsForPackageOwnedByOrg` seems to [be an async](https://github.com/octokit/plugin-rest-endpoint-methods.js/blob/ec6e43c0d80cfbb1fa7768b7dc43368abaa63ad2/src/generated/method-types.ts#L7817) function so should be awaited
   
```suggestion
            const response = await github.rest.packages.getAllPackageVersionsForPackageOwnedByOrg({
```